include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -796,6 +798,76 @@ struct __simple_write_to_idx
     }
 };
 
+template <typename _Predicate>
+struct __gen_mask
+{
+    template <typename _InRng>
+    bool
+    operator()(_InRng&& __in_rng, std::size_t __idx) const
```

Comment:
Since we aren't forwarding this forwarding reference anywhere, would it be better accept `__in_rng` as a const l-value ref: `const _InRng& __in_rng`?

Reply:
yes, good point. I'll check the other similar helpers to see if there are others to adjust too.

Reply:
cleaned this up here, and in the previous PR.  I'll scan through the rest of the PRs as well.